pkgs/shells/zsh/zimfw/default.nix

```
@@ -0,0 +1,42 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zimfw";
+  version = "v1.11.0";
+  src = fetchFromGitHub {
+    owner = "zimfw";
+    repo = pname;
+    rev = version;
+    ## zim only needs this one file to be installed.
+    sparseCheckout = [ "zimfw.zsh" ];
+    sha256 = "sha256-BmzYAgP5Z77VqcpAB49cQLNuvQX1qcKmAh9BuXsy2pA=";
+  };
+  strictDeps = true;
+  dontConfigure = true;
+  dontBuild = true;
+  dontPatch = true;
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r $src/zimfw.zsh $out/
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out
    cp -r $src/zimfw.zsh $out/
    
    runHook postInstall
  '';
```

When overriding phases like this, remember to still run the pre/post hooks